src/yyjson.c

```diff
@@ -5794,6 +5794,15 @@ yyjson_doc *yyjson_read_file(const char *path,
 #undef return_err
 }
 
+bool unsafe_yyjson_read_number(const char *dat, usize len, void *val) {
+    u8 *hdr, *end;
+    const char *msg;
+    if (unlikely(!dat || !len)) return false;
+    hdr = (u8 *)dat, end = hdr + len;
+    if (!read_number(&hdr, NULL, false, (yyjson_val *)val, &msg)) return false;
+    return hdr == end;
```

Comment:
Here we can return the end of the string directly, which provides more information to the caller (as `strtod()` does).

Reply:
I've updated the commit to return the end pointer instead of a boolean.